strfry/plugins/tests/broadcast_vanish_requests.test.ts

```diff
@@ -1,27 +1,68 @@
 import { assertEquals } from "https://deno.land/std@0.181.0/testing/asserts.ts";
 import { buildEvent, buildInputMessage } from "./test.ts";
 import { createBroadcastVanishRequests } from "../broadcast_vanish_requests.ts";
+import { PubkeyCache } from "../pubkey_cache.ts";
 import type { Event } from "https://raw.githubusercontent.com/planetary-social/strfry-policies/refs/heads/nos-changes/mod.ts";
 
 class RedisMock {
   called: boolean = false;
+  zset: Map<string, number> = new Map();
 
   async xadd(streamKey: string, id: string, event: Event): Promise<void> {
     this.called = true;
   }
+
+  async zrevrange(key: string, start: number, stop: number): Promise<string[]> {
+    const entries = Array.from(this.zset.entries());
+    entries.sort((a, b) => b[1] - a[1]);
+    const sliced = entries.slice(start, stop + 1);
+    return sliced.map(([pubkey, _score]) => pubkey);
+  }
+
+  async zcard(key: string): Promise<number> {
+    return this.zset.size;
+  }
+
+  async zremrangebyrank(
```

Comment:
Should these `z.*` functions be somewhere other than the test file since they are called from the main code?

Reply:
This is just a test mock for the Deno redis library.